app/src/main/java/com/ieeevit/todoandroid/adapters/TodoTaskAdapter.java

```diff
@@ -75,20 +87,25 @@ public void onClick(View view) {
             itemView.setOnLongClickListener(new View.OnLongClickListener() {
                 @Override
                 public boolean onLongClick(View view) {
+                    if(getSelectedIndex()==-1) {
+                        checkedPosition = getAdapterPosition();
+                        checkedImageView.setVisibility(View.VISIBLE);
+                        notifyItemChanged(getAdapterPosition());
+                    }
                     // Set the visibility of the checkedImageView by checking the checked index
                     // Don't forget to call notifyItemChanged() with the checkedPosition as argument
-                    return false;
+                    return true;
                 }
             });
         }
     }
 
     /**
      * Constructor for the adapter.
-     * @param context
      * @param tasks the list of TodoTasks.
      */
-    public TodoTaskAdapter(ArrayList<TodoTask> tasks) {
+    public TodoTaskAdapter(ArrayList<TodoTask> tasks, Context context) {
+        this.context=context;
```

Comment:
I think it would be best not to pass something as heavy as `Context` to the adapter. In `onCreateViewHolder()`, `parent.getContext()` is being used to retrieve the context, so we don't need it to be passed to the constructor anymore.